io/io/src/TBufferJSON.cxx

```diff
@@ -3294,10 +3294,21 @@ void TBufferJSON::WriteFastArray(const Bool_t *b, Int_t n)
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Write array of Char_t to buffer
+///
+/// Normally written as JSON string, but if string includes 0 in the middle
+/// or some special characters, uses regular array. From array size 1000 it
+/// will be automatically converted into base64 coding
 
 void TBufferJSON::WriteFastArray(const Char_t *c, Int_t n)
 {
-   JsonWriteFastArray(c, n, "Int8", &TBufferJSON::JsonWriteConstChar);
+   Bool_t need_blob = false, has_zero = false;
+   for (int i=0;i<n;++i)
+      if (!c[i]) has_zero = true; else if (has_zero || (c[i] < 10)) need_blob = true;
```

Comment:
Could you split this line in several lines (easier to set break points when debugging)

Reply:
Done